compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -169,8 +169,10 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
     s eq defn.newArrayMethod
   }
 
-  def isBox(sym: Symbol): Boolean = Erasure.Boxing.isBox(sym)
-  def isUnbox(sym: Symbol): Boolean = Erasure.Boxing.isUnbox(sym)
+  def isBox(sym: Symbol): Boolean = Erasure.Boxing.isBox(sym) && !isInUnit(sym)
```

Comment:
Why not change the definition of Erasure.Boxing.isBox directly?

Reply:
Erasure does transformations on Unit box inbox based on Boxing.isBox. If I add that condition directly there erasure starts failing.

Reply:
ok